pkg/osbuild/stage.go

```diff
@@ -28,3 +28,7 @@ func (s *Stage) MountOSTree(osName, ref string, serial int) {
 	ostreeMount := NewOSTreeDeploymentMount(name, osName, ref, serial)
 	s.Mounts = append(s.Mounts, *ostreeMount)
 }
+
+type FileChanger interface {
+	FilesChanged() []string
+}
```

Comment:
Nitpick: I had a slight problem with naming the functions using "files" in the original PR, because this is about paths (symlinks, directories, files) in general, not just files. Before, the code was limited to bootc, so it was not a big deal from my PoV. Since we are now generalizing the solution, which is great, let's use something more descriptive (maybe `PathsChanged()`)?

At the end of the day, I can live with the naming as is, just wanted to point this out.

Reply:
Yeah, I did this as a quick draft to show the concept, its incomplete, I need to implement the other stages like chmod/chown and iirc some more if we want to continue. I like your name better, I think `PathsChanged()` it is :) 

The alternative would be just he simple approach Alex outlined in https://github.com/osbuild/images/pull/2037#issuecomment-3575925113 ie just always relabeling /etc, /var if we have post stages. I really like the simplicity of the appraoch but also worry about the precision (long term). So I'm a bit on the fence about it and wanted to hear the others first :) But happy to properly flesh this out.